### PR TITLE
Feature wavewindow

### PR DESCRIPTION
**Pull request by jdmcd**
@EricKuck sets up a (rough) skeleton for what some of the classes should look like

### CODE REVIEW

**Comment by jdmcd** [comment]
@EricKuck ready for code review

File: Wave/Wave/Section.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//
+//  Created by Jimmy McDermott on 6/27/16.
+//  Copyright © 2016 Blue Line Labs. All rights reserved.
+//
+
+import Foundation
+
+public class Section: SectionType {
+    public var name: String = ""
+    var settings = [Setting<Any>]()
```
Comment (EricKuck) [inline Wave/Wave/Section.swift]: Any -> AnyObject so we can store it.

File: Wave/Wave/Section.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public func numberOfRows() -> Int {
+        return settings.count
+    }
+    
+    public func tableViewCellIdentifier(indexPath: NSIndexPath) -> String {
+        return settings[indexPath.row].cellIdentifier
+    }
+    
+    public class func addRow() -> Section {
+        return Section()
```
Comment (EricKuck) [inline Wave/Wave/Section.swift]: Why is a new section being returned here? I was thinking we'd just return self so it ends up being more of a builder pattern.

File: Wave/Wave/SectionType.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
+//
+//  SectionType.swift
+//  Wave
+//
+//  Created by Jimmy McDermott on 6/27/16.
+//  Copyright © 2016 Blue Line Labs. All rights reserved.
```
Comment (EricKuck) [inline Wave/Wave/SectionType.swift]: Our name is BlueLine Labs. Should prob update this in all your copyrights (and Xcode)

File: Wave/Wave/SettingType.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+import Foundation
+
+protocol SettingType {
+    associatedtype T
+    
+    var defaultValue: T? { get set }
+    var key: String { get set }
+}
```
Comment (EricKuck) [inline Wave/Wave/SettingType.swift]: Newline, please

File: Wave/WaveExample/AppDelegate.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import UIKit
+import Wave
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+
+
```
Comment (EricKuck) [inline Wave/WaveExample/AppDelegate.swift]: -newline

**Pull request merged**